### PR TITLE
fix:removed the upstash redis and upstash ratelimit

### LINKED ISSUE #143: Remove Upstash Depedency
Currently some parts are reliant on Upstash (mainly the ratelimiting). To be truly self-hostable we should get rid of this.

This would include the following

- Remove the `@upstash/redis` dependency

and either
- Remove the `@upstash/ratelimit` dependency
- Add a new rate limiting library

or

- Make the `@upstash/ratelimit` dependency work without the upstash redis client

### PR DESCRIPTION
**Pull request by rajeshj11**
/claim #143 
/fixes #143 
1. removed the @upstash/redis and @upstash/rate-limit
2. added a new package for handling the rate limit(rate-limiter-flexible)

## Summary by CodeRabbit

- **New Features**
  - Improved rate limiting using a new system based on `RateLimiterRedis`.

- **Chores**
  - Updated package dependencies, replacing `@upstash/ratelimit` and `@upstash/redis` with `rate-limiter-flexible`.

### CODE REVIEW

**Comment by rajeshj11** [comment]
@cstrnt please review

File: apps/web/src/api/routes/v1_event.ts [hunk 1]
```diff
@@ -1,14 +1,31 @@
...
+import { redis } from "server/db/redis";
+
+export const rateLimiter = new RateLimiterRedis({
+  storeClient: redis,
+  keyPrefix: "rateLimiter",
+  points: 10, // Number of points
+  duration: 10, // Per 10 seconds
+});
+
+export const checkRateLimit = async (ip: string) => {
```
Comment (cstrnt) [inline apps/web/src/api/routes/v1_event.ts:20]: I don't think this should be exported here :)

**Review by cstrnt — CHANGES_REQUESTED** [review]
This looks really good! I think it would be amazing if you could extract the `checkRateLimit` function with the ratelimiter to it's own file/module so that we can potentially re-use this :) Otherwise this looks amazing!

File: apps/web/src/api/routes/v1_event.ts (hunk 1 above)
Comment (rajeshj11) [reply apps/web/src/api/routes/v1_event.ts:20]: Will make changes accordingly and update you.

**Comment by rajeshj11** [comment]
> This looks really good! I think it would be amazing if you could extract the `checkRateLimit` function with the ratelimiter to it's own file/module so that we can potentially re-use this :) Otherwise this looks amazing!

can I move these functions to 
/home/rajesh/github/abby/apps/web/src/api/routes/utils.ts

**Comment by cstrnt** [comment]
> > This looks really good! I think it would be amazing if you could extract the `checkRateLimit` function with the ratelimiter to it's own file/module so that we can potentially re-use this :) Otherwise this looks amazing!
> 
> can I move these functions to /home/rajesh/github/abby/apps/web/src/api/routes/utils.ts

Would prefer `server/common/ratelimit.ts`

**Comment by rajeshj11** [comment]
> > > This looks really good! I think it would be amazing if you could extract the `checkRateLimit` function with the ratelimiter to it's own file/module so that we can potentially re-use this :) Otherwise this looks amazing!
> > 
> > 
> > can I move these functions to /home/rajesh/github/abby/apps/web/src/api/routes/utils.ts
> 
> Would prefer `server/common/ratelimit.ts`

done

**Comment by cstrnt** [comment]
@rajeshj11 the `pnpm-lock.yaml` is out of sync. Could you please run `pnpm i` locally and check in this file?

**Comment by rajeshj11** [comment]
Ok

**Review by cstrnt — APPROVED** [review]

**Pull request merged**